### PR TITLE
Provide more information for nested types in NavigateTo to help disambiguate them

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/7432

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @rchande

**Comment by CyrusNajmabadi** [comment]
Looks like this now: 

image

File: src/Workspaces/Core/Portable/FindSymbols/DeclaredSymbolInfo.cs [hunk 1]
```diff
@@ -105,7 +106,12 @@ internal struct DeclaredSymbolInfo
             parameterCount = Math.Min(parameterCount, (byte)MaxFlagValue);
             typeParameterCount = Math.Min(typeParameterCount, (byte)MaxFlagValue);
 
-            _flags = (uint)kind | ((uint)accessibility << 4) | ((uint)parameterCount << 8) | ((uint)typeParameterCount << 12);
+            _flags =
+                (uint)kind |
+                ((uint)accessibility << 4) |
+                ((uint)parameterCount << 8) |
+                ((uint)typeParameterCount << 12) |
+                ((isNestedType ? 1u : 0u) << 13);
```
Comment (sharwell) [inline src/Workspaces/Core/Portable/FindSymbols/DeclaredSymbolInfo.cs]: ❕  typeParameterCount is 4 bits wide.

😢  literals

Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/FindSymbols/DeclaredSymbolInfo.cs]: nice!

**Comment by sharwell** [comment]
@CyrusNajmabadi What happens when you have something like this, and search for `Nested`?

```
class A {
  class Outer {
    class Nested { }
  }
}

class B {
  class Outer {
    class Nested { }
  }
}
```

**Comment by CyrusNajmabadi** [comment]
@rchande These tests don't run locally under @Pilchie 's wpf runner.  So i'm just using jenkins to see what fails and i'll update those tests.

**Comment by CyrusNajmabadi** [comment]
@sharwell that looks like this:

image

**Comment by CyrusNajmabadi** [comment]
Switched format string slightly.  I like the new style better.

**Comment by CyrusNajmabadi** [comment]
I'm thinking of changing all nested symbols (types and members) to just say: 

"In XXX (project ...)"

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by CyrusNajmabadi** [comment]
retest windows_release_vs-integration_prtest please

**Comment by CyrusNajmabadi** [comment]
retest windows_release_unit32_prtest  please

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_vs-integration_prtest please

**Pull request merged**